editoast/src/views/projects.rs

```diff
@@ -126,18 +127,19 @@ async fn check_image_content(db_pool: Data<DbConnectionPool>, document_key: i64)
 }
 
 #[derive(Debug, Clone, Serialize, ToSchema)]
-pub struct ProjectWithStudies {
+pub struct ProjectWithStudyCount {
     #[serde(flatten)]
     project: Project,
-    studies_count: i64,
+    studies_count: u64,
 }
 
-impl ProjectWithStudies {
-    pub fn new_from_project(project: Project, studies_count: i64) -> Self {
-        Self {
+impl ProjectWithStudyCount {
+    async fn new(conn: &mut DbConnection, project: Project) -> Result<Self> {
```

Comment:
It doesn't look like a pattern I see often to have an async-fallible function called `new`. At the very least, because it's fallible, I would call it `try_new`. As for the `async` part, maybe `init`or `fetch` would be better? So something like `try_init` or `try_fetch`?